### PR TITLE
Feature/add authorization rules

### PR DESCRIPTION
**Pull request by belalgamal46**
# Description
This pull request adds several new features and updates to the project, including:

- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
  - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

# Changes Made

- Installed CanCanCan gem and added role column to users table
- Created a delete button for comments for specific users using CanCanCan
- Added ability model to define user permissions
- Updated controller helper functions and added CanCanCan rescue
- Updated likes controller
- Added after_destroy method and updated update_posts_counter method
- Added destroy method in posts controller
- Added destroy routes for posts and comments
- Added delete post button with styles
- Added admin method in the user model
- Added delete post form
- Updated likes model
- Fixed some bugs and linters

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -13,4 +13,8 @@ class User < ApplicationRecord
   def latest_three_posts
     posts.order(created_at: :desc).limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (Reem-lab) [inline app/models/user.rb:19]: - kudos  that you create  user.admin? method inside the user model 🔥💪🏻✅🎉

**Review by Reem-lab — APPROVED** [review]
# Hi @belalgamal46 and @Gmanlove,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the funtionality for adding the  authorization rules💯
- [ ] Followed Gitflow 🔥
- [ ] No Linters error ✅



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (belalgamal46) [reply app/models/user.rb:19]: @Reem-lab thanks a lot yeah I really worked on myself 😅😅

Comment (Gmanlove) [reply app/models/user.rb:19]: Thanks

**Pull request merged**